naga/src/front/wgsl/lower/mod.rs

```diff
@@ -2077,6 +2087,21 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
                 // reference is required, the Load Rule is not applied.
                 match self.expression_for_reference(expr, ctx)? {
                     Typed::Reference(handle) => {
+                        if let &crate::Expression::Access { base, .. }
+                        | &crate::Expression::AccessIndex { base, .. } = ctx.get(handle)
+                        {
+                            if let Some(ty) = resolve_inner!(ctx, base).pointer_base_type() {
+                                if matches!(
+                                    *ty.inner_with(&ctx.module.types),
+                                    crate::TypeInner::Matrix { .. }
```

Comment:
I don't think this should apply to matrices. I don't know why Tint is also restricting matrices, the only mention of matrices in the spec discussion is:

> We have to determine what to do with matrices (which may or may not match what we're doing with vectors)

_from https://github.com/gpuweb/gpuweb/issues/1931#issuecomment-935180135_

Since it's not mentioned again, I assume someone checked and they were fine.

@jrprice did you guys find issues with matrices? If so, we should add the restriction to the spec.

Reply:
Taking the address of a matrix component requires you to take the address of a vector component, since you have to go through a column vector to get there. So this case would also be covered by the existing WGSL spec restriction.

Reply:
I must have messed something up when I was looking at the treatment of matrices before, maybe I wasn't careful enough to distinguish address-of-column from address-of-element.

In any case, I've changed it to disallow address-of vector component only.